go/arrow/flight/flightsql/driver/driver.go

```diff
@@ -37,34 +38,50 @@ import (
 )
 
 type Rows struct {
-	schema        *arrow.Schema
-	records       []arrow.Record
-	currentRecord int
-	currentRow    int
+	schema           *arrow.Schema
+	recordChan       chan arrow.Record
+	currentRecord    arrow.Record
+	currentRecordMux sync.Mutex
```

Comment:
do we actually need this mutex? I thought that the `Rows` object isn't used by multiple goroutines concurrently?

Reply:
mutex removed.
wasn't really consistent.
I think the new organization is leaner & safer.